### DISCUSSION TITLE
BlueRetro version ( v1.9.1 - Dec 27 2023 23:38:25) Issue

### DISCUSSION THREAD

**Discussion by parfuar** [darthcloud/BlueRetro#871] [Q&A]
Hey,
When I use the blueretro I connect my controller (original PS4 or "parallel" PS3 is not an original controller) it works, but during the game, the game is "paused" automatically sometimes, so I continue playing and the same situation arises again ...

On the "serial monitor" I saw these cases...
I compile and send it to the ESP-32 and even after configuring it as PS2 on the website it appears the system says SATURN "System detected: 12: SATURN" and I have two opcode errors.

���[0;33mW (29) boot: Unicore bootloader[0m
[0;32mI (272) cpu_start: Unicore app[0m
[0;32mI (272) cpu_start: Single core mode[0m
[0;32mI (280) cpu_start: Pro cpu start user code[0m
[0;32mI (280) cpu_start: cpu freq: 240000000 Hz[0m
[0;32mI (280) cpu_start: Application information:[0m
[0;32mI (280) cpu_start: Project name:     BlueRetro[0m
[0;32mI (281) cpu_start: App version:      v1.9.1[0m
[0;32mI (282) cpu_start: Compile time:     Dec 27 2023 23:38:25[0m
[0;32mI (282) cpu_start: ELF file SHA256:  8823339a9...[0m
[0;32mI (283) cpu_start: ESP-IDF:          v5.2-dev-3318-g8fc8f3f479[0m
[0;32mI (284) cpu_start: Min chip rev:     v1.0[0m
[0;32mI (284) cpu_start: Max chip rev:     v3.99 [0m
[0;32mI (285) cpu_start: Chip rev:         v3.1[0m
[0;32mI (286) cpu_start: App cpu up.[0m
[0;32mI (286) heap_init: Initializing. RAM available for dynamic allocation:[0m
[0;32mI (287) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM[0m
[0;32mI (288) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM[0m
[0;32mI (288) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM[0m
[0;32mI (289) heap_init: At 3FFCDF90 len 00012070 (72 KiB): DRAM[0m
[0;32mI (290) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
[0;32mI (291) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM[0m
[0;32mI (292) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM[0m
[0;32mI (292) heap_init: At 40078000 len 00008000 (32 KiB): IRAM[0m
[0;32mI (293) heap_init: At 4009F51C len 00000AE4 (2 KiB): IRAM[0m
[0;32mI (294) heap_init: At 3FF80068 len 00001F98 (7 KiB): RTCRAM[0m
[0;32mI (296) spi_flash: detected chip: generic[0m
[0;32mI (296) spi_flash: flash io: dio[0m
[0;32mI (297) esp_core_dump_uart: Init core dump to UART[0m
[0;32mI (297) esp_core_dump_uart: Init core dump to UART[0m
[0;32mI (298) main_task: Started on CPU0[0m
[0;32mI (298) main_task: Calling app_main()[0m
 App main at 40082B80
 App init at 40082B90
 APP_CPU RESET: 0
 APP_CPU CLKGATE: 0
 APP_CPU STALL: 0
 APP_CACHE_CTRL: 00000030
 APP CPU STACK PTR: 3FFBCED8
 GPIO[19]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[21]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[22]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[25]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[32]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[33]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[34]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[35]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
 GPIO[39]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[19]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[21]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[22]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[25]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[32]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[33]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[34]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[35]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[39]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 Detected system : 12: SATURN <------------------------------------------             I changed to PS2 by website bluereto.io
 GPIO[35]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:2
 GPIO[36]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:2
 GPIO[27]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[16]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[26]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[23]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[18]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[5]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[3]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[33]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[25]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[22]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[21]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 GPIO[19]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 bt_host_load_bdaddr_from_file: No BDADDR on SD. Using ESP32's MAC
[0;32mI (348) BTDM_INIT: BT controller compile version [ec4ac65][0m
[0;32mI (348) BTDM_INIT: Bluetooth MAC: 24:dc:c3:99:ef:92[0m
[0;32mI (358) phy_init: phy_version 4780,16b31a7,Sep 22 2023,20:42:16[0m
Coex register Wi-Fi channel change btdm cb faild  <------------------------------------------
 gid_update_sys: SATURN  <------------------------------------------
 bt_hci_cmd_reset
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_features
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_version_info
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_bd_addr
 BT_HCI_EVT_CMD_COMPLETE
 local_name: BlueRetro_SATURN_EF92
 local_bdaddr: 24:DC:C3:99:EF:92
 bt_hci_cmd_read_buffer_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_class_of_device
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_name
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_voice_setting
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_num_supported_iac
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_current_iac_lap
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_conn_accept_timeout
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_supported_commands
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_ssp_mode
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_inquiry_mode
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_inquiry_rsp_tx_pwr_lvl
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_ext_features
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_stored_link_key
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_page_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_page_scan_type
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_le_host_supp
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_read_wl_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_clear_wl
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_delete_stored_link_key
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_class_of_device
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_local_name
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_auth_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_mask
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_inquiry_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_type
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_timeout
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_hold_mode_act
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_scan_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_default_link_policy
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_read_buffer_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_att_set_le_max_mtu 247
 bt_hci_cmd_le_set_adv_param
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_adv_data
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_scan_rsp_data
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_adv_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x200C error: 0x0C retry: 0 <------------------------------------------
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x0403 error: 0x0C retry: 0 <------------------------------------------
 mc_restore: restore sucessful!
[0;32mI (888) gpio: GPIO[0]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0 [0m
 internal_flag_init: External adapter
[0;32mI (888) gpio: GPIO[2]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
[0;32mI (888) gpio: GPIO[4]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
[0;32mI (888) main_task: Returned from app_main()[0m
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_CONN_REQUEST
 bt_hci_cmd_accept_conn_req
 Page dev: 0 type: 0 bdaddr: 64:D4:BD:9B:51:08
 BT_HCI_EVT_CMD_STATUS
 BT_HCI_EVT_ROLE_CHANGE
 BT_HCI_EVT_CONN_COMPLETE
 dev: 0 acl_handle: 0x0080
 bt_hci_cmd_le_set_adv_disable
 bt_hci_cmd_remote_name_request
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_STATUS
 BT_L2CAP_CONN_REQ
 BT_HCI_EVT_REMOTE_NAME_REQ_COMPLETE:
 bt_type_update: dev: 0 type: 1 subtype: 0
 bt_hci_cmd_exit_periodic_inquiry
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_passive
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_switch_role
 dev: 0 type: 1:0 PLAYSTATION(R)3 Controller
 BT_L2CAP_CONF_REQ
 BT_L2CAP_CONF_REQ
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_STATUS
 BT_HCI_EVT_ROLE_CHANGE
 BT_L2CAP_CONN_REQ
 BT_L2CAP_CONF_REQ
 BT_L2CAP_CONF_RSP
 bt_hid_ps3_init
 bt_hid_ps3_init_callback

I believe these two problems are the reason the game is automatically paused
The environment used in "ubuntu 20.04 desktop" (ubuntu-20.04.6-desktop-amd64 - new installation)

Thanks

**Comment by darthcloud** [comment]
Can you get the log from the beginning while it work as PS2.

Look like the adapter crash a some point and is re-detected as another system.
That's kind of possible since the system IOs aren't idle like at boot time.

The real issue to figure out is why it reboot in the first place, so would need the log from beginning.

You could use the playstation specific firmware to avoid the detection problem.

**Reply by parfuar** [reply]
That's what I did... I compiled it directly on "PS2 system" and it was resolved
Do you know why there are these opcode lines? example -> "opcode: 0x0403 error: 0x0C retry: 0" - (ESP works normally). 
However, I created some functions where the PS2 turns on and off with the wireless controller and the ESP is able to read the PS2 state (ON or OFF) 😁😁
